docs/contrib/Logger.rst

```diff
@@ -1,174 +1,310 @@
+
+   Incomplete:
+   Warning: {} does not have viaDef aligned with layer.
+
+Make any assumptions or use of default values explicit:
+
+.. code-block:: text
+
+   Correct:
+   No net slacks found.
+   Timing-driven mode disabled.
+
+   Incomplete, missing default:
+   Utilization exceeds 100%.
+
+Use simple language, avoid repetitions:
+
+.. code-block:: text
+
+   Correct:
+   Missing orientation for cell $cell_ref
+
+   Incorrect:
+   No orientation available for orientation of $cell_ref.
 
 Message Types
--------------
+~~~~~~~~~~~~~
+
+OpenROAD supports the following levels of severity through the logger:
+Report, Debug, Information, Error, Warning and Critical.
 
 Report
-~~~~~~
+++++++
 
 Reports are tool output in the form of a report to the user. Examples
-are timing paths, or power analysis results. Tool reports that use
-‘printf’ or c++ streams should use the report message API instead.
+include timing paths or power analysis results.
+
+Example Report message:
+
+.. code-block:: text
+
+    Path startpoint: $startpoint
 
 Debug
-~~~~~
++++++
 
 Debug messages are only of use to tool developers and not to end users.
 These messages are not shown unless explicitly enabled.
 
 Information
-~~~~~~~~~~~
++++++++++++
 
 Information messages may be used for reporting metrics, quality of
-results, or program status to the user. Any messages which indicate
+results, or program status to the user. Any message which indicates
 runtime problems, such as potential faulty input or other internal
 program issues, should be issued at a higher status level.
 
-Example messages for this status level:
+Example Information messages:
 
-::
+.. code-block:: text
 
    Number of input ports: 47
    Running optimization iteration 2
    Current cell site utilization: 57.1567%
 
 Warning
-~~~~~~~
++++++++
 
 Warnings should be used for indicating atypical runtime conditions that
 may affect quality, but not correctness of the output. Any conditions
 that affect correctness should be issued at a higher status level.
 
-Example warning messages:
+Example Warning messages:
 
-::
+.. code-block:: text
 
    Core area utilization is greater than 90%. The generated cell placement may not be routable.
+
    14 outputs are not constrained for max capacitance.
-   Pin ‘A[0]’ on instance ‘mem01’ does not contain antenna information and will not be checked for antenna violations.
+
+   Pin 'A[0]' on instance 'mem01' does not contain antenna information and will not be checked for antenna violations.
 
 Error
-~~~~~
++++++
 
 Error messages should be used for indicating correctness problems.
 Problems with command arguments are a good example of errors. Errors
-exit the current command by throw an error that can be caught in a Tcl
+exit the current command by throwing an exception that can be caught in a Tcl
 command script. Errors that occur while reading a command file stop
```

Comment:
... throwing an exception that is converted to an error in Tcl.

Reply:
this got garbled.

Reply:
fixed